### ISSUE TITLE
Stravalib2.0 -> Maintain Strava's meta-summary-detailed type hierarchy

### ISSUE THREAD

**Issue by jsamoocha** [stravalib/stravalib#499]
This should prevent response objects with many `None` fields and make it more explicit to users what level of detail they'll get from an API call.

- [x] Update client methods to ensure the correct detail level is returned
- [x] Check the model for instances where we override stuff from other than detailed-level objects

Depends on #498 

**Update:** As a reminder: this will also close #198

**Comment by lwasser** [comment]
it will be a lot less confusing to have fewer of these empty response objects. Will this also then address the comment in our discussion surrounding laps? i still haven't had a chance to look into that but i have noticed empty returns.

**Comment by jsamoocha** [comment]
Yes, exactly. We discussed this some time ago, and this recent issue triggered me to fix this now that we remove that `BackwardCompatibilityMixin` and introduce breaking changes anyway.

**Comment by lwasser** [comment]
perfect! i tested this and it's working well lately via your latest pr!! so much better!

**Comment by lwasser** [comment]
## Client methods that return iterators

- [ ] get_activities (#505)
- [ ] get_athlete_koms
- [ ] get_segment (#509)
- [ ] get_Starred_segment (#509)
- [ ] get_club_activities()
- [ ] get_club_members
- [ ] get_club_admins 
- [ ] get_activity_comments ()
- [ ] get_activity_kudos
- [ ] get_activity_photos
- [ ] get_activity_laps ()
- [ ] get_related_Activities (REMOVE) 
- [ ] get_friend_Activities (REMOVE) #todo create new issue to remove deprecated methods

- [ ] get_starred_segments
- [ ] get_athlete_starred_segments
- [ ] get_segment_efforts
- [ ] get_routes ()


------
- [ ] get_athlete ?

**Comment by lwasser** [comment]
i believe we have completed this issue via the recent endpoint checks and now that the Mixin is removed a well. closing. please reopen if I am missing something we haven't implemented!
